.travis.yml

```diff
@@ -1,69 +1,130 @@
+      name: pip, Linux, py3.6
+      env: MB_PYTHON_VERSION=3.6
+    - os: osx
+      name: pip, macOS, py2.7
+      language: generic
+      env: MB_PYTHON_VERSION=2.7
     - os: osx
+      name: pip, macOS, py3.6
       language: generic
+      env: MB_PYTHON_VERSION=3.6
+    # Use CMake to setup and install libecl
+    - os: linux
+      name: cmake, Linux, py2.7
       python: 2.7
       env:
         - MB_PYTHON_VERSION=2.7
-        - TEST_SUITE=""
+        - LEGACY_INSTALL=1
+    - os: linux
+      name: cmake, Linux, py3.6
+      python: 3.6
+      env:
+        - MB_PYTHON_VERSION=3.6
+        - LEGACY_INSTALL=1
     - os: osx
+      name: cmake, macOS, py2.7
+      language: generic
+      env:
+        - MB_PYTHON_VERSION=2.7
+        - LEGACY_INSTALL=1
+    - os: osx
+      name: cmake, macOS, py3.6
       language: generic
-      python: 3.6
       env:
         - MB_PYTHON_VERSION=3.6
-        - TEST_SUITE=""
-
-addons:
-  apt:
-    packages:
-    - liblapack-dev
+        - LEGACY_INSTALL=1
 
 before_install:
-  - unset -f pushd
-  - unset -f popd
+  # Test the C stuff before we do anything in pip mode
+  - |
+    if [ -z "${LEGACY_INSTALL}" ]; then
+      mkdir build && pushd build
+      cmake .. -DCMAKE_BUILD_TYPE=Release -DBUILD_TESTS=ON
+      cmake --build .
+      ctest --output-on-failure
+      popd && rm -rf build
+    fi
+
+  - git clone https://github.com/matthew-brett/multibuild
   - source multibuild/common_utils.sh
   - source multibuild/travis_steps.sh
   - before_install
 
 install:
-  - pip install -r requirements.txt
+  - |
+    if [ -z "${LEGACY_INSTALL}" ]; then
+      build_wheel . $PLAT
+    else
+      pip install -r requirements.txt
+
+      mkdir build
+      pushd build
+      cmake .. -DBUILD_TESTS=ON                               \
+               -DENABLE_PYTHON=ON                             \
+               -DBUILD_APPLICATIONS=ON                        \
+               -DINSTALL_ERT_LEGACY=ON                        \
+               -DERT_USE_OPENMP=ON                            \
+               -DCMAKE_C_FLAGS='-Werror=all'                  \
+               -DCMAKE_CXX_FLAGS='-Werror -Wno-unused-result'
+      make
+      sudo make install
+      popd
+    fi
 
 script:
-  - mkdir build
-  - pushd build
-  - cmake .. -DBUILD_TESTS=ON
-             -DENABLE_PYTHON=ON
-             -DBUILD_APPLICATIONS=ON
-             -DINSTALL_ERT_LEGACY=ON
-             -DERT_USE_OPENMP=ON
-             -DCMAKE_C_FLAGS='-Werror=all'
-             -DCMAKE_CXX_FLAGS='-Werror -Wno-unused-result'
-  - make
-  - sudo make install
-  - which python
-  - export PYTHONPATH="/usr/local/lib/python$MB_PYTHON_VERSION/site-packages:/usr/local/lib/python$MB_PYTHON_VERSION/dist-packages:$PYTHONPATH"
-  - python -c "import sys; print('\n'.join(sys.path))"
-  - set -e; python -c "import ecl"; set +e
-  - ctest --output-on-failure $TEST_SUITE
-  - popd
+  - |
+    if [ -z "${LEGACY_INSTALL}" ]; then
+      install_run $PLAT
```

Comment:
Where does `install_run` come from?

Reply:
Is it multi build magic?

Reply:
multibuild magic. It's in `travis_steps.sh`